Description: Fix to give buttons variable width according to string
 This patch is based on ideas from Alexey Osipov to fix the bug regarding
 failure of winff to assign the proper space for strings. This patch
 only fixes the buttons as they are the most visible and easiest to fix.
 Further fixing is awaiting response from upstream.
Author: Paul Gevers <paul@climbing.nl> / Alexey Osipov
Bug: http://code.google.com/p/winff/issues/detail?id=47
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/winff/+bug/695040
Forwarded: not-needed

Index: winff-1.3.1/unit1.lfm
===================================================================
--- winff-1.3.1.orig/unit1.lfm	2011-01-08 17:26:59.000000000 +0100
+++ winff-1.3.1/unit1.lfm	2011-01-09 21:55:10.600793001 +0100
@@ -465,7 +465,10 @@
       Height = 53
       Top = 0
       Width = 60
+      Align = alLeft
+      AutoSize = True
       Caption = 'Add'
+      Constraints.MinWidth = 60
       Glyph.Data = {
         36090000424D3609000000000000360000002800000018000000180000000100
         2000000000000009000064000000640000000000000000000000FFFFFF00FFFF
@@ -552,7 +555,10 @@
       Height = 53
       Top = 0
       Width = 60
+      Align = alLeft
+      AutoSize = True
       Caption = 'Remove'
+      Constraints.MinWidth = 60
       Glyph.Data = {
         36090000424D3609000000000000360000002800000018000000180000000100
         2000000000000009000064000000640000000000000000000000FFFFFF00FFFF
@@ -639,7 +645,10 @@
       Height = 53
       Top = 0
       Width = 60
+      Align = alLeft
+      AutoSize = True
       Caption = 'Clear'
+      Constraints.MinWidth = 60
       Glyph.Data = {
         36090000424D3609000000000000360000002800000018000000180000000100
         2000000000000009000064000000640000000000000000000000FFFFFF00FFFF
@@ -725,8 +734,11 @@
       Left = 200
       Height = 53
       Top = 0
-      Width = 70
+      Width = 60
+      Align = alLeft
+      AutoSize = True
       Caption = 'Play'
+      Constraints.MinWidth = 60
       Glyph.Data = {
         36090000424D3609000000000000360000002800000018000000180000000100
         2000000000000009000064000000640000000000000000000000FFFFFF00FFFF
@@ -812,8 +824,11 @@
       Left = 344
       Height = 53
       Top = 0
-      Width = 70
+      Width = 60
+      Align = alLeft
+      AutoSize = True
       Caption = 'Convert'
+      Constraints.MinWidth = 60
       Glyph.Data = {
         36090000424D3609000000000000360000002800000018000000180000000100
         2000000000000009000064000000640000000000000000000000FFFFFF00FFFF
@@ -899,18 +914,22 @@
       Left = 938
       Height = 52
       Top = 1
-      Width = 125
+      Width = 0
       Align = alRight
       BevelOuter = bvNone
       ClientHeight = 52
-      ClientWidth = 125
+      ClientWidth = 0
       TabOrder = 5
+    end
       object btnOptions: TBitBtn
-        Left = 40
+        Left = 740
         Height = 53
         Top = -1
         Width = 60
+        Align = alRight
+        AutoSize = True
         Caption = 'Options'
+        Constraints.MinWidth = 60
         Glyph.Data = {
           36090000424D3609000000000000360000002800000018000000180000000100
           2000000000000009000064000000640000000000000000000000FFFFFF00FFFF
@@ -992,13 +1011,15 @@
         OnClick = mitShowOptionsClick
         TabOrder = 0
       end
-    end
     object btnPreview: TBitBtn
       Left = 272
       Height = 53
       Top = 0
       Width = 70
+      Align = alLeft
+      AutoSize = True
       Caption = 'Preview'
+      Constraints.MinWidth = 60
       Glyph.Data = {
         36090000424D3609000000000000360000002800000018000000180000000100
         2000000000000009000064000000640000000000000000000000FFFFFF00FFFF
